src/api/entitlements.js

```diff
@@ -175,6 +175,31 @@ export class Entitlements {
   ack() {
     this.ackHandler_(this);
   }
+
+  /**
+   * Converts the whole set of entitlements into a single SKU.
+   * Returns the first SKU that enables this page.
+   * TODO: come up with a better way of picking the right SKU when there are
+   * multiple
+   * @return {?string}
+   */
+  getSingleEntitlement() {
+    if (!this.entitlements) {
+      return null;
+    }
+    for (let i = 0; i < this.entitlements.length; i++) {
+      const ent = this.entitlements[i];
+      const sku = ent.getSku();
+      if (!sku) {
+        continue;
+      }
+      if (this.product_ && !ent.enables(this.product_)) {
+        continue;
+      }
+      return sku;
+    }
+    return null;
```

Comment:
Do we want to throw an exception instead?

Reply:
You could certainly not have any entitlements so an exception would not be appropriate.